pkg/apis/core/validation/shoot_test.go

```diff
@@ -422,6 +422,34 @@ var _ = Describe("Shoot Validation Tests", func() {
 			})))),
 		)
 
+		DescribeTable("addons validation",
+			func(purpose core.ShootPurpose, version string, allowed bool) {
+				shootCopy := shoot.DeepCopy()
+				shootCopy.Spec.Purpose = &purpose
+				shootCopy.Spec.Kubernetes.Version = version
+				shootCopy.Spec.Kubernetes.KubeAPIServer.EnableBasicAuthentication = pointer.BoolPtr(false)
+
+				errorList := ValidateShoot(shootCopy)
+
+				if allowed {
+					Expect(errorList).To(BeEmpty())
+				} else {
+					Expect(errorList).To(ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Type":  Equal(field.ErrorTypeForbidden),
+						"Field": Equal("spec.addons"),
+					}))))
+				}
+			},
+			Entry("should allow addons on evaluation shoots with version >= 1.22", core.ShootPurposeEvaluation, "1.22.0", true),
+			Entry("should forbid addons on testing shoots with version >= 1.22", core.ShootPurposeTesting, "1.22.0", false),
```

Comment:
Could you add a test with pre-release k8s version, e.g `1.22.0-alpha.1`?

Reply:
Done (in a new commit)
/squash